### DISCUSSION TITLE
Helping Patching Iosevka SS12 Variant Font

### DISCUSSION THREAD

**Discussion by kaykhan** [ryanoasis/nerd-fonts#1176] [General]
Hi friends,

i like to use a variant of Iosevka called Iosevka-ss12.

The current patched nerd font of Iosevka https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts/Iosevka is not the variant im looking for. 

**Iosevka SS12**

![image](https://user-images.githubusercontent.com/5202527/233792096-80e6828f-04ec-4a7f-bde9-490e6d48f70e.png)

**Iosevka Nerd Font**
https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts/Iosevka
![image](https://user-images.githubusercontent.com/5202527/233792151-5c0bd4f2-83d6-4e1b-be01-4f42c2b22677.png)

There are some difference within `Iosevka SS12` that i like, that are not apart of the existing patched nerd font`Iosevka Nerd Font`.

---

I patched the Iosevka SS12 font succssfully using the fontforge tool, but i noticed there are still a few problems.

1. Still has some Chinese characters (the patch did fix some)
2. Small icons

![image](https://user-images.githubusercontent.com/5202527/233792288-54d07ef9-375a-42da-ac82-4d6b176787f5.png)

i would really like some help with patching this font.

**Comment by Finii** [comment]
Well, usually you do not need to self-patch just to get hold to one stylistic set.
The `Iosevka Nerd Font` contains all Stylistic Sets (that includes SS12), you just need to activate it in your application:

![image](https://user-images.githubusercontent.com/16012374/233829551-b89bb894-2a1f-4f1a-954f-69863dda3171.png)

There is a good howto here:
https://github.com/tonsky/FiraCode/wiki/How-to-enable-stylistic-sets

I personally used https://github.com/twardoch/fonttools-opentype-feature-freezer to create a 'dumb' font for 'dump' applications.

If you really want to self-patch the SS12 variant directly and encounter problems we would need screenshots or logs of the process.
Usually small icons means you specified `-s` on the command line. Problems with Chinese glyphs ... might not be any. This also depends on the `font-patcher` version you used and on command line parameters. Usually if you see Chinese glyphs where you expected a symbol that means you use an obsolete codepoint. Check your symbol's codepoints with the https://www.nerdfonts.com/cheat-sheet.

Here is an example, just use the new codepoint. There are several projects out there that help you with the codepoint update.

![image](https://user-images.githubusercontent.com/16012374/233829879-6a0da2be-3e2c-475a-8b51-e3febcc0019d.png)
_Use `F0826` instead of `FD24`_
